node/taskcommand.ts

```diff
@@ -33,12 +34,16 @@ export class TaskCommand {
                     var val = this.properties[key];
                     if (val) {
                         cmdStr += key + '=' + val + ';';
+                        if(key === 'secret' && val === 'true') {
+                            containsSecret = true;
+                        }
                     }                    
                 }
             }
         }
 
-        cmdStr += ']' + this.message;
+        cmdStr += ']';
+        cmdStr += containsSecret? '********' : this.message;
```

Comment:
Not sure I like baking in the concept of a 'secret' property into the command class.  This is a generic parser.

Why is this change needed?  When is the toString of the cmd leaking to a log?  And why wouldn't registering it as a secret suffice via setVariable?